src/population.cpp

```diff
@@ -344,6 +345,26 @@ std::string Population::_dynamicsAttributeDataType(const std::string& name) cons
     return _getDataType(impl_->getDynamicsAttributeDataSet(name), name);
 }
 
+template <>
+Selection Population::filterAttribute(const std::string& name,
+                                      std::function<bool(const std::string)> pred) const {
+    auto dtype = impl_->getAttributeDataSet(name).getDataType();
+    if (dtype != HighFive::AtomicType<std::string>()) {
```

Comment:
Could this not be generalized to
```suggestion
    if (dtype != HighFive::AtomicType<T>()) {
```
And merged with below?

Reply:
I need to manually specialize it, since the macros don't specialize std::string for other reasons.

Reply:
Alrighty